CIP-1853/CIP-1853.md

```diff
@@ -0,0 +1,54 @@
+---
+CIP: 1853
+Title: HD (Hierarchy for Deterministic) Stake Pool Cold Keys for Cardano
+Authors: Rafael Korbas <rafael.korbas@vacuumlabs.com>
+Comments-URI: <TBD>
+Status: Draft
+Type: Standards
+Created: 2020-12-14
+License: CC-BY-4.0
+---
+
+## Abstract
+
+[CIP-1852](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/CIP-1852.md) establishes how Shelley-era hierarchical deterministic (HD) wallets should derive their keys. This document is a follow-up of this CIP specifying how stake pool cold keys should be derived.
+
+## Motivation
+
+(Hierarchical) deterministic derivation of cold keys allows for their restorability from a seed and most importantly, their management on hardware wallet devices which in turn mitigates man-in-the middle attacks to which pool operators would otherwise be vulnerable if they managed their stake pool cold keys on a device not specifically hardened against alteration of the data to be signed/serialized without operator's explicit consent.
+
+## Specification
+
+Using `1853'` as the purpose field, we define the following derivation path structure for stake pool cold keys.
+
+```
+m / purpose' / coin_type' / usecase / cold_key_index'
+```
+
+Example: `m / 1853' / 1815' / 0 / 0'`
+
+Here the `usecase` is currently fixed to `0`.
+
+Given that stake pool cold keys are cryptographically the same as wallet keys already covered in CIP-1852, the master node and subsequent child keys derivation **MUST** be implemented in the same way as specified for wallets in CIP-1852.
+
+## Rationale
+
+### Why introducing a new purpose?
+
+Stake pools are not wallets and the core concept of "accounts" is not applicable to them, nor are they supposed to be related to user's wallet in any meaningful way. Therefore treating stake pool cold keys as another "chain" within CIP-1852 specification would rather be a deviation from CIP-1852 than its logical extension. Hence we establish a separate purpose and path structure for stake pool cold keys, having their specifics and differences from standard "wallet" keys in mind.
+
+### Why keeping `coin_type` in the path?
+
+`coin_type` is kept in order to remain consistent with the "parent" CIP-1852 and also to leave space for the possibility that some Cardano hard-fork/clone in the future would reuse this specification to derive their own stake pool cold keys.
+
+### `usecase` field
+
+Similarly as we have the `chain` path component in CIP-1852 paths, it's plausible that there can be multiple "kinds" of stake pools in the future even though currently there's only one. One such example of a potential future extension of this CIP could be pools managed by a group of operators instead of a single operator, for which a separate key chain may make sense both from logical and security perspective.
```

Comment:
"key chain" is not  defined here

Reply:
It should be explained why usecase is not hardened.

Reply:
"key chain" is a term already used in CIP-11 (https://github.com/cardano-foundation/CIPs/blob/931af2b15a20da6d8b0a678b1b05aa1ee39b3ec5/CIP-0011/CIP-0011.md) so I'd say it's common knowledge for the target audience

Reply:
regarding usecase non-hardened derivation, I just added a paragraph about it below

Reply:
I gave this only a cursory glance so far, and "key chain" was mentioned without any reference or definition, so I found it rather confusing in the context of dealing with cold _keys_.

CIP-11 only mentions "staking key chain", not chain in general, so it would be best not to mention it at all.
CIP-1852 is OK, though it only mentions "chain", never a "key chain". That's exactly what we have here now.

I would use 'multiple varieties' instead of 'multiple "kinds"'.

Reply:
reworded "kinds" to "varieties" and rewrote the paragraph to talk about a separate "set of keys", drawing concrete similarities with wallet keys